dask/dataframe/core.py

```diff
@@ -1654,6 +1654,37 @@ def count(self, axis=None, split_every=False):
                 result.divisions = (self.columns.min(), self.columns.max())
             return result
 
+    @derived_from(pd.DataFrame)
+    def mode(self, axis=0, dropna=True, split_every=False):
+        if axis == 1:
+            meta = self._meta_nonempty.count(axis=axis)
+            result = map_partitions(
+                M.mode,
+                self,
+                meta=meta,
+                token=self._token_prefix + "mode",
+                axis=axis,
+                dropna=dropna,
+                enforce_metadata=False,
+            )
+            return result
+        else:
+
+            value_count_series = self.reduction(
+                chunk=M.value_counts,
+                aggregate=M.sum,
+                split_every=split_every,
+                chunk_kwargs={"dropna": dropna},
+            )
+
+            max_val = value_count_series.max(skipna=dropna)
+            mode_series = (
+                value_count_series[value_count_series == max_val]
+                .index.to_series()
+                .reset_index(drop=True)
+            )
+            return mode_series
```

Comment:
I currently don't sort the mode values as pandas does, but thinking more about it, I don't believe sorting would require any additional shuffling since by the time the sorting would be done, all the data is in a single partition (if I understand the reduction step correctly).  I'd be happy to add sorting unless there is a reason not too. @TomAugspurger, any thoughts?

Reply:
I recommend that you create a function that does all of the final assembly bits and then return something like the following:

```python
return self.reduction(
    chunk=M.value_counts,
    combine=M.sum,
    aggregate=_mode_aggregate,  # (you'll make this function somewhere in this module, see _count_aggregate for an example
    ...
)
```

Reply:
I added the _mode_aggregate function as recommended.